swift_scality_backend/diskfile.py

```diff
@@ -548,3 +553,11 @@ def delete(self, timestamp):
                           called externally only by the `ObjectController`
         """
         self._filesystem.del_object(self._name)
+
+    @property
+    def timestamp(self):
+        if self._metadata is None:
+            raise swift.common.exceptions.DiskFileNotOpen()
+        return swift.common.utils.Timestamp(self._metadata.get('X-Timestamp'))
```

Comment:
This approach is wrong. First:
```python
>>> object(1)
Traceback (most recent call last):
  File "<ipython-input-1-eac988317dd7>", line 1, in <module>
    object(1)
TypeError: object() takes no parameters
```
So, if `swift.common.utils.Timestamp` doesn't exist this code would fail.

I'm not very fond of monkey-patching `swift.common.utils`. What I suggest instead is something along the lines of:

```python
if hasattr(swift.common.utils, 'Timestamp'):
    @property
    def timestamp(self):
        ...

    data_timestamp = timestamp
```

and remove the monkey-patch higher up.